ProcessLib/ThermalTwoPhaseFlowWithPP/ThermalTwoPhaseFlowWithPPMaterialProperties.h

```diff
@@ -0,0 +1,122 @@
+                            TwoPhaseFlowWithPPMaterialProperties>
+            two_phase_material_model,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            specific_heat_capacity_solid,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            specific_heat_capacity_water,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            specific_heat_capacity_air,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            specific_heat_capacity_vapor,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            thermal_conductivity_dry_solid,
+        std::unique_ptr<MaterialLib::Fluid::FluidProperty>
+            thermal_conductivity_wet_solid,
+        std::unique_ptr<MaterialLib::Fluid::WaterVaporProperties>
+            water_vapor_properties);
+
+    double getSpecificHeatCapacitySolid(const double p, const double T) const;
+    double getSpecificHeatCapacityWater(const double p, const double T) const;
+    double getSpecificHeatCapacityAir(const double p, const double T) const;
+    double getSpecificHeatCapacityVapor(const double p, const double T) const;
+    double getThermalConductivityDrySolid(const double p, const double T) const;
+    double getThermalConductivityWetSolid(const double p, const double T) const;
+    /// Calculates the unsaturated heat conductivity
+    double calculateUnsatHeatConductivity(double const t,
+                                          ProcessLib::SpatialPosition const& x,
+                                          double const Sw,
+                                          double const lambda_pm_dry,
+                                          double const lambda_pm_wet) const;
+    /// water vapor saturation pressure
+    double calculateSaturatedVaporPressure(const double T) const;
+    /// partial water vapor pressure in nonwetting phase
+    /// Kelvin equation
+    double calculateVaporPressureNonwet(const double pc, const double T,
+                                        const double rho_mass_h2o) const;
+    /// Derivative of SaturatedVaporPressure in terms of T
+    double calculateDerivativedPsatdT(const double T) const;
+    /// Derivative of partial vapor pressure in terms of T
+    double calculateDerivativedPgwdT(const double pc, const double T,
+                                     const double rho_mass_h2o) const;
+    /// Derivative of partial vapor pressure in terms of PC
+    double calculateDerivativedPgwdPC(const double pc, const double T,
+                                      const double rho_mass_h2o) const;
+    ///
+    double calculatedRhoNonwetdT(const double p_air_nonwet,
+                                 const double p_vapor_nonwetconst, double pc,
+                                 const double T,
+                                 const double rho_mass_h2o) const;
+
+    MaterialLib::TwoPhaseFlowWithPP::TwoPhaseFlowWithPPMaterialProperties*
+    getTwoPhaseMaterialModel()
```

Comment:
The function could be const, I believe. Why not return a reference, since this object would own the material model anyway. This would save checking for nullptr on the callers side...